### ISSUE TITLE
Uncaught TypeError: Cannot read property 'concat' of undefined

### ISSUE THREAD

**Issue by guyellis** [survivejs/webpack_react#81]
Happens when you run the Chap 06 version of the app from here: `project_source/06_react_and_flux/kanban_app` after `npm i` and then `npm start`

**Comment by bebraw** [comment]
I have a feeling this might have something to do with `localStorage`. Having invalid data there could easily cause this. I'll need to make the implementation more robust.

* Can you check out its content and let me know what you have there? I'm primarily interested in the content of `notes`.
* What browser are you using? Just trying to rule out a browser specific issue here. I know the implementation needs to grow to work around Safari/IE glitches.

**Comment by guyellis** [comment]
I'm using Chrome on Ubuntu. The problem persists after I clear local storage. Seems that Chrome is in a state that it can't recover from. I tried on Firefox and that works without this error. I hadn't used Firefox before for this.

**Comment by bebraw** [comment]
@guyellis Can you see what line triggers the error?

We might want to patch `init` of `NoteStore`. Ie. instead of

```javascript
  init(data) {
    this.setState(data || {notes: []});
  }
```

we could add a check against data to make sure it contains an array of notes. If it doesn't, we'll just revert to the default value. I'm thinking something along

```javascript
  init(data) {
    var d = {notes: []};

    if(_.isObject(data) && _.isArray(data.notes)) {
      d = data;
    }

    this.setState(d);
  }
```

Now, even if the initial data gets corrupted somehow it won't crash. Can you try it with a code similar to this?

**Comment by guyellis** [comment]
I just tried that and your suggestion worked. Obviously had to also `npm i lodash --save` and reference it.

I then stashed the changes and tried to reproduce the problem again but was unable to. I didn't check to see if the problem still existed immediately before trying the changes so I can't be sure if the changes then "fixed" something in chrome or if some other chrome usage during the day "fixed" the problem.

It might also be easier to do:

```javascript
  init(data) {
    var d = _.isArray(_.get(data, 'notes')) ? data : {notes: []};

    this.setState(d);
  }
```
